src/ranged.cpp

```diff
@@ -2695,10 +2697,78 @@ void target_ui::apply_aim_turning_penalty()
 
 void target_ui::action_switch_mode()
 {
-    relevant->gun_cycle_mode();
-    if( relevant->gun_current_mode().flags.count( "REACH_ATTACK" ) ) {
-        relevant->gun_cycle_mode();
+    uilist menu;
+    menu.settext( _( "Select preferences" ) );
+    const std::pair<int, int> aim_modes_range = std::make_pair( 0, 100 );
+    const std::pair<int, int> firing_modes_range = std::make_pair( 100, 200 );
+
+
+    // list aiming modes
+    menu.addentry( -1, false, 0, "  " + std::string( _( "Default aiming mode" ) ) );
+    menu.entries.back().force_color = true;
+    menu.entries.back().text_color = c_cyan;
+
+    for( auto it = aim_types.begin(); it != aim_types.end(); ++it ) {
+        const bool is_active_aim_mode = aim_mode == it;
+        const std::string text = ( it->name.empty() ? _( "Immediate" ) : it->name ) +
+                                 ( is_active_aim_mode ? _( " (active)" ) : "" );
+        menu.addentry( aim_modes_range.first + std::distance( aim_types.begin(), it ),
+                       true, MENU_AUTOASSIGN, text );
+        if( is_active_aim_mode ) {
+            menu.entries.back().text_color = c_light_green;
+        }
+    }
```

Comment:
Aiming does not work & does not matter when firing vehicle-mounted weapons (`mode == TargetMode::TurretManual`)

Reply:
Could you expand this a bit?
It seems you can't reach either switching ammo or firing mode when doing manual turret aiming, (but you also can't do it on master ), not sure if you're supposed to from your comment or not

Reply:
>It seems you can't reach either switching ammo or firing mode when doing manual turret aiming, (but you also can't do it on master)

Not quite.
When both aiming a gun & doing manual turret aiming, it's possible to switch ammo & switch firing mode
https://github.com/CleverRaven/Cataclysm-DDA/blob/119a79311942adec5821206d66d3b8808ec9d961/src/ranged.cpp#L2174-L2177
https://github.com/CleverRaven/Cataclysm-DDA/blob/119a79311942adec5821206d66d3b8808ec9d961/src/ranged.cpp#L2031-L2037
But mounted guns handle recoil in their own way and don't use `ACT_AIM` activity (that allows the whole multi-turn aiming thing), so aiming (and, consequently, switching aiming mode) for them is not enabled:
https://github.com/CleverRaven/Cataclysm-DDA/blob/119a79311942adec5821206d66d3b8808ec9d961/src/ranged.cpp#L2178-L2181

Then, you change `action_switch_mode` from "switch firing mode only" to "switch firing and/or aiming modes". Now, when player tries to switch firing mode on a turret, your menu pops up and prompts to switch aiming mode as well.

Reply:
Right I also followed up on those input calls but I can't seem to reach
either the menu with the patch or do any switching when manning the
turret on master

May be I'm misunderstanding how to get to it;
My setup is random car, installed  turret mount + fn-scar L mounted, loaded fn-scar
Unwielding weapon, stand on controls tile, press `f`
I now see I'm aiming with turret - but:
Pressing either `m` or `s` doesn't do anything at all with the turret
^ the following is on master
testing with the patch - doing the same thing my menu doesn't pop up

Am I trying it wrong?

Reply:
> fn-scar L mounted

Hmm, I can't find how to mount an fn-scar L on a vehicle. I spawned a humvee and mounted a Browning Automatic Rifle instead - it has both burst & single-fire
![image](https://user-images.githubusercontent.com/60584843/95664971-d72ad900-0b54-11eb-8c72-ee597cf84e8e.png)

>stand on controls tile

Try the tile with the turret mount - you should see aiming UI for manual turret firing
![image](https://user-images.githubusercontent.com/60584843/95665003-11947600-0b55-11eb-9da1-83fe6dbfec7d.png)
Pressing `s` switches firing mode between `semi-auto (1)` and `auto (3)`

This is on 595373dfdc

Reply:
>Try the tile with the turret mount - you should see aiming UI for manual turret firing

That was it, sorry I assumed they're the same, it should now hide titles without choices,
Could also hide if there's only semi auto, but the sidepanel still lists the hotkey - i'll leave
it for now

Reply:
>I assumed they're the same

Yeah, the whole manual turret aiming vs manual turret firing is confusing as heck, both in code and in the game.

Btw, it looks like `SWITCH_AIM` hotkey is now functionally same as `SWITCH_MODE`. The status check in
https://github.com/CleverRaven/Cataclysm-DDA/blob/0f16482201bf31a73c839eaec5c45749b2fa51bc/src/ranged.cpp#L2038-L2042
was mostly a cosmetic thing for UI feedback (it doesn't matter which aim you use if your gun can't fire that far / you can't aim there - so the aim switching was disabled, and corresponding entry grayed out), but the new mode selection menu bypasses this check anyway when invoked via `SWITCH_MODE`...

Reply:
Hm... That seems to make sense when you cycle modes on a
specific target and look at the % going up or down but does
it now when you just open a menu and set a default?

I think just letting the player set it whenever they like, merging
both actions into one if(..) path (so people don't lose their favorite
keybind) and editing the panel to hide one of them. That leaves
an unnecessary keybind though.

That sounds sensible?

Reply:
IMO yes. Or just remove `SWITCH_AIM`, and add `m` to `SWITCH_MODE` as alternative to `s` and `F`. Or repurpose it to to work as "cycle aiming mode" in the mode selection menu (I believe #39048 did that with `"` key)

Reply:
Talked it out on discord a bit - seems m wasn't really popular anyway so it had to go

Looks like keybinds can't be migrated for existing configs - so if someone really likes to
switch with m and they already have the config set they'll have to add it themselves